ui/src/app/pages/appAnalysis/apps-list.tsx

```diff
@@ -9,40 +10,137 @@ import { Button } from 'app/components/buttons';
 import { FadeBox } from 'app/components/containers';
 import { FlexColumn, FlexRow } from 'app/components/flex';
 import { ListPageHeader } from 'app/components/headers';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
 import colors from 'app/styles/colors';
-import { withCurrentWorkspace } from 'app/utils';
+import { reactStyles, withCurrentWorkspace } from 'app/utils';
 import { AnalyticsTracker } from 'app/utils/analytics';
+import { APP_LIST, JUPYTER_APP } from 'app/utils/constants';
+import { WorkspacePermissionsUtil } from 'app/utils/workspace-permissions';
+
+const styles = reactStyles({
+  fadeBox: {
+    margin: 'auto',
+    marginTop: '1rem',
+    width: '95.7%',
+  },
+  startButton: {
+    paddingLeft: '0.5rem',
+    height: '2rem',
+    backgroundColor: colors.secondary,
+  },
+  appsLabel: {
+    color: colors.primary,
+    fontWeight: 600,
+    fontSize: '14px',
+    lineHeight: '24px',
+    paddingBottom: '0.5rem',
+  },
+});
 
 export const AppsList = withCurrentWorkspace()((props) => {
   const { workspace } = props;
+  const [selectedApp, setSelectedApp] = useState('');
+  const [showAppModal, setShowAppModal] = useState(false);
+  const [showJupyterModal, setShowJupyterModal] = useState(false);
+
+  const displayJupyterModal = () => {
+    setShowAppModal(false);
+    setShowJupyterModal(true);
+  };
+
+  const canWrite = (): boolean => {
+    return WorkspacePermissionsUtil.canWrite(props.workspace.accessLevel);
+  };
+
+  const showApplicationModal = () => {
```

Comment:
if the app type is jupyter, this includes a call to `setShowAppModal(false)` which I think will be confusing.  Is there a better name for this?

Reply:
great, thank you